Decisions.TruBot/Steps/WaitRunBotStep.cs

```diff
@@ -0,0 +1,120 @@
+using System.Net.Http.Json;
+using Decisions.TruBot.Api;
+using Decisions.TruBot.Data;
+using DecisionsFramework;
+using DecisionsFramework.Data.ORMapper;
+using DecisionsFramework.Design.ConfigurationStorage.Attributes;
+using DecisionsFramework.Design.Flow;
+using DecisionsFramework.Design.Flow.Mapping;
+using DecisionsFramework.Design.Flow.StepImplementations;
+using DecisionsFramework.Design.Properties;
+using DecisionsFramework.ServiceLayer;
+
+namespace Decisions.TruBot.Steps
+{
+    [Writable]
+    [AutoRegisterStep("Run Bot and Wait", "Integration/TruBot/Bot")]
+    [ShapeImageAndColorProvider(null, TruBotSettings.TRUBOT_IMAGES_PATH)]
+    public class WaitRunBotStep : IAsyncStep, IDataConsumer
+    {
+        private const string PATH_DONE = "Done";
+        private const string TRUBOT_RESPONSE = "TruBot Status Response";
+        
+        private const string BOT_ID = "BotId";
+        
+        [WritableValue]
+        private string overrideBaseUrl;
+
+        [PropertyClassification(0, "Override Base URL", "Override Settings")]
+        public string OverrideBaseUrl
+        {
+            get => overrideBaseUrl;
+            set => overrideBaseUrl = value;
+        }
+
+        public void Start(StepStartData data)
+        {
+            int botId = data.Data[BOT_ID] as int? ?? 0;
+
+            DateTime startTime = DateTime.Now;
+
+            string baseUrl = ModuleSettingsAccessor<TruBotSettings>.GetSettings().GetBotUrl(OverrideBaseUrl);
+            string runUrl = $"{baseUrl}/RunBot";
+
+            TruBotAuthentication auth = new TruBotAuthentication
+            {
+                Token = ModuleSettingsAccessor<TruBotSettings>.GetSettings().Token,
+                Sid = ModuleSettingsAccessor<TruBotSettings>.GetSettings().Sid
+            };
+
+            TruBotProcess botProcess;
+            TruBotResponse response = new TruBotResponse();
+            try
+            {
+                BotIdRequest inputs = new BotIdRequest();
+                inputs.BotId = botId;
+                
+                JsonContent content = JsonContent.Create(inputs);
+                
+                string result = TruBotRest.TruBotPost(runUrl, auth, content);
+                response = TruBotResponse.JsonDeserialize(result);
+
+                ORM<TruBotRecordedBot> recordedBotOrm = new ORM<TruBotRecordedBot>();
```

Comment:
Can we move this code to a helper class?
Ideally we can name the two methods that do ORM work, something that helps clarify what is happening in the code.

`TruBotRecordedBot recBot = TruBotRecordedBot.Create(response.BotId, response.BotName, startTime);`
would be a good pattern to clean this up I think.